### ISSUE TITLE
EFI boot using grub does not work for debian-live-9.3.0-cinnamon

### ISSUE THREAD

**Issue by eaglerainbow** [mbusb/multibootusb#248]
As a side-effect of the test at https://github.com/mbusb/multibootusb/issues/242#issuecomment-354503020 I figured out yet another issue

**Environment**
* Status as of branch `devel`, commit 22226a4.
* USB3.0 stick
* formatted with `parted` using
   * `mklabel gpt`
   * one partition, fat32
* installing image as of https://cdimage.debian.org/debian-cd/current-live/amd64/iso-hybrid/debian-live-9.3.0-amd64-cinnamon.iso, no second image installed
* System is started in Lenovo L560 machine (USB3.0 port only)
* BIOS configuration setup to only allow EFI-booting.

**Observation**
Grub is booted up in menu mode (thanks!), but image cannot be started.

**Expected behavior**
Grub starts the image (grub-loopback happening)

**Further insights**
I little local debugging reveiled the following facts to me:
* The menu defines the command `configfile  /multiboot/debian-live-9.3.0-amd64-cinnamon/boot/grub/loopback.cfg`, which is fine as it exists.
* However, that file itself reads: 
```
source /multiboot/debian-live-9.3.0-amd64-cinnamon/grub/grub.cfg
```
Note here that the file does **not** ⚠️ `/multiboot/debian-live-9.3.0-amd64-cinnamon/boot/grub/grub.cfg` (which would exist), but the `boot/` folder is missed out.
* Manually entering `source /multiboot/debian-live-9.3.0-amd64-cinnamon/boot/grub/grub.cfg` is also not successful, as it returns the error message:
```
error: file '/multiboot/grub/font.pf2` not found. 
error: no video mode activated
```
Please note that `/multiboot/debian-live-9.3.0-amd64-cinnamon/boot/grub/font.pf2` exists on the USB stick.

**Comment by mbusb** [comment]
Did you try bootable USB on a fresh if USB. Not on the one created earlier
by multibootusb. I have modified grub.cfg file as well.

On 30-Dec-2017 3:21 AM, "eaglerainbow" <notifications@github.com> wrote:

As a side-effect of the test at #242 (comment)
<https://github.com/mbusb/multibootusb/issues/242#issuecomment-354503020> I
figured out yet another issue

*Environment*

   - Status as of branch devel, commit 22226a4
   <https://github.com/mbusb/multibootusb/commit/22226a40fcd06f8950e893daaef6fd8804ce0ecc>
   .
   - USB3.0 stick
   - formatted with parted using
      - mklabel gpt
      - one partition, fat32
   - installing image as of https://cdimage.debian.org/
   debian-cd/current-live/amd64/iso-hybrid/debian-live-9.3.0-
   amd64-cinnamon.iso
   <https://cdimage.debian.org/debian-cd/current-live/amd64/iso-hybrid/debian-live-9.3.0-amd64-cinnamon.iso>,
   no second image installed
   - System is started in Lenovo L560 machine (USB3.0 port only)
   - BIOS configuration setup to only allow EFI-booting.

*Observation*
Grub is booted up in menu mode (thanks!), but image cannot be started.

*Expected behavior*
Grub starts the image (grub-loopback happening)

*Further insights*
I little local debugging reveiled the following facts to me:

   - The menu defines the command configfile /multiboot/debian-live-9.3.0-
   amd64-cinnamon/boot/grub/loopback.cfg, which is fine as it exists.
   - However, that file itself reads:

source /multiboot/debian-live-9.3.0-amd64-cinnamon/grub/grub.cfg

Note here that the file does *not* ⚠️ /multiboot/debian-live-9.3.0-
amd64-cinnamon/boot/grub/grub.cfg (which would exist), but the boot/ folder
is missed out.

   - Manually entering source /multiboot/debian-live-9.3.0-
   amd64-cinnamon/boot/grub/grub.cfg is also not successful, as it returns
   the error message:

error: file '/multiboot/grub/font.pf2` not found.
error: no video mode activated

Please note that /multiboot/debian-live-9.3.0-amd64-cinnamon/boot/grub/font.
pf2 exists on the USB stick.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/mbusb/multibootusb/issues/248>, or mute the thread

**Comment by eaglerainbow** [comment]
> Did you try bootable USB on a fresh if USB. Not on the one created earlier by multibootusb. I have modified grub.cfg file as well. 

Yes, I had zero-ed out the USB stick and created a new fat32 using `mkfs.vfat`. Afterwards I ran the entire copy process over it. I cannot imagine that something old could still be standing...

**Comment by shinji-s** [comment]
> Note here that the file does not ⚠️ read /multiboot/debian-live-9.3.0-amd64-cinnamon/boot/grub/grub.cfg (which would exist), but the boot/ folder is missed out.

https://github.com/mbusb/multibootusb/commit/170b8dcf5835b4b4a23f5d65c1cf922f73e5f43b should have fixed this issue.
In my environment debian-live-9.4.0-amd64-cinnamon boots up albeit with error messages about missing fonts and splash screen.
